### ISSUE TITLE
Upgrading to v2, error *4848 port is already used* in zulu 

### ISSUE THREAD

**Issue by hantsy** [actions/setup-java#158]
**Platform:**
- [x] Ubuntu
- [ ] macOS
- [ ] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

Follow the [migrate guide](https://github.com/actions/setup-java/blob/main/docs/switching-to-v2.md), after upgrading to v2, all my ci using Glassfish failed due to a *4848 port is already used*,

https://github.com/hantsy/jakartaee9-starter-boilerplate/blob/b3e51fcd857768c05e6104ea940f0ecd283c4698/.github/workflows/it-with-arq-glassfish-managed.yml#L22

but switch back to v1, work again.

https://github.com/hantsy/jakartaee9-starter-boilerplate/blob/master/.github/workflows/it-with-arq-glassfish-managed.yml#L22

**Comment by dmitry-shibanov** [comment]
Hello @hantsy. I think it's not related to `setup-java` task, because I forked your repository and could reproduce the issue with `setup-java@v1`. 
 - [it-with-arq-glassfish-managed](https://github.com/dmitry-shibanov/jakartaee9-starter-boilerplate/runs/2294663587?check_suite_focus=true)
 - [it-with-arq-glassfish-remote](https://github.com/dmitry-shibanov/jakartaee9-starter-boilerplate/runs/2294376295?check_suite_focus=true)

**Comment by hantsy** [comment]
The tests in my master is now passed.

You can check the commit log, I am trying to upgrade to v2, all failed. and return back to v1, worked.

**Comment by hantsy** [comment]
The same problem when using v2 with Payara(a fork of Glassfish), check the details.
https://github.com/hantsy/cargotracker/runs/2311107552#step:6:1543

**Comment by hantsy** [comment]
Not sure the workflows in the repository cached the old JDK, when returning to the v1,  some former successful workflows in  https://github.com/hantsy/cargotracker failed now.

**Comment by dmitry-shibanov** [comment]
Hello @hantsy. We investigated the issue and don't think it's something task related. As we can see the issue reproduces on both `v1` and `v2` versions:
- `v2`: [v2 passed](https://github.com/hantsy/jakartaee9-starter-boilerplate/runs/2274403037?check_suite_focus=true) and [v2 failed](https://github.com/hantsy/jakartaee9-starter-boilerplate/runs/2283730069?check_suite_focus=true)
- `v1`: [v1 failed](https://github.com/hantsy/jakartaee9-starter-boilerplate/runs/2339792580?check_suite_focus=true)

**Comment by hantsy** [comment]
So strange, also reported to Payara, also do not know the reason.

And the most strange is the GlassFish v6.0/Jakarta EE 9 testing codes failed now.

**Comment by hantsy** [comment]
Till now, almost all Payara/GlassFish related workflows which worked well in the past months or years are failed in all of my repositories.

*  https://github.com/hantsy/jakartaee8-starter-boilerplate
*  https://github.com/hantsy/jakartaee9-starter-boilerplate
*  https://github.com/hantsy/cargotracker/ 

I was afraid the port conflict is caused by the workflow execution in parallel,  but in the above projects, also configured multi workflows for  WildFly, OpenLiberty, these workflows never failed.

**Comment by maxim-lobanov** [comment]
@hantsy , every workflow is run on the separate clean VM that is isolated from other VMs so any parallel conflicts are eliminated.
As for the port 4848, the command works as expected. But before the steps `Run Glassfish Server` and ` Run integration test with -Parq-payara-managed` port is free. So looks like something goes wrong during this step.

**Comment by hantsy** [comment]
You can check the commit log,  before 5, April before the setup-java v2 is released, the workflow for `glassfish-remote` is like this. it worked well for serval months.

https://github.com/hantsy/jakartaee9-starter-boilerplate/blob/6c5a20a8335cc9cc94e763d1020ac0e4668b034b/.github/workflows/it-with-arq-glassfish-remote.yml

I am not sure why `glassfish-managed` worked again in the newest version in https://github.com/hantsy/jakartaee9-starter-boilerplate, it failed most of time in the past 2 weeks(after 5, April).

**Comment by hantsy** [comment]
Created a new Circle CI config, https://github.com/hantsy/cargotracker/blob/master/.circleci/config.yml#L29, the scripts is 100% same as the Github actions payara-managed workflow.

The Circle CI environment never reports *4848 is already in use* error.  And this error nerver occured on local system.

I also tried to print the port info in some steps, https://github.com/hantsy/cargotracker/blob/master/.github/workflows/it-with-arq-payara-managed.yml#L54, but Github actions will exit quickly when executing this scripts. So I do not know which program is using 4848 at all.

**Comment by maxim-lobanov** [comment]
I suggest trying add the following step to the failed workflow right before invocation of your tool:
```
- run: sudo netstat -lpn
```
It should list all busy ports.

I can't reproduce the issue right after build starts, port is free:
```
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:8084            0.0.0.0:*               LISTEN      753/mono            
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      547/systemd-resolve 
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      718/sshd: /usr/sbin 
tcp        0      0 127.0.0.1:41399         0.0.0.0:*               LISTEN      715/containerd      
tcp6       0      0 :::22                   :::*                    LISTEN      718/sshd: /usr/sbin 
udp        0      0 127.0.0.53:53           0.0.0.0:*                           547/systemd-resolve 
udp        0      0 10.1.0.4:68             0.0.0.0:*                           527/systemd-network 
udp        0      0 127.0.0.1:323           0.0.0.0:*                           676/chronyd         
udp6       0      0 ::1:323                 :::*                                676/chronyd         
raw6       0      0 :::58                   :::*                    7           527/systemd-network
```

**Comment by hantsy** [comment]
I am crazy, really not sure why it happened.

**Comment by hantsy** [comment]
@maxim-lobanov  In these two days, there are some PRs merged into the above repos, it seems everything becomes as before.  The broken workflows(due to 4848 port issues) are done successfully.

**Comment by maxim-lobanov** [comment]
That sounds really weird because neither `actions/setup-java` nor Hosted runners (both successful and failed builds are run on 20210412.1 image version) were updated. I wonder if it could it be an issue with some kind of dependencies

**Comment by hantsy** [comment]
@maxim-lobanov maybe a system package that JDK is dependent on caused this issue.  Thanks for your patience. I close this issue.

**Comment by hantsy** [comment]
@maxim-lobanov  This issue occurred again, https://github.com/hantsy/jakartaee9-starter-boilerplate/runs/4853696050?check_suite_focus=true#step:4:2266
